src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -1019,16 +1019,28 @@ void AwtWindow::_RepositionSecurityWarning(void* param)
     JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
     RepositionSecurityWarningStruct *rsws =
-        (RepositionSecurityWarningStruct *)param;
+        static_cast<RepositionSecurityWarningStruct *>(param);
     jobject self = rsws->window;
 
-    PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
-    AwtWindow *window = (AwtWindow *)pData;
+    AwtWindow *window = nullptr;
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        delete rsws;
+        return;
+    } else {
+        window = (AwtWindow *) JNI_GET_PDATA(self);
+        if (window == nullptr) {
+            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
+            env->DeleteGlobalRef(self);
+            delete rsws;
+            return;
+        }
+    }
 
     window->RepositionSecurityWarning(env);
 
-  ret:
```

Comment:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.